tests/config/nextflow.config

```
@@ -0,0 +1,77 @@
+params {
+    outdir = "output/"
+    publish_dir_mode = "copy"
+    enable_conda = false
+    singularity_pull_docker_container = false
+    max_cpus   = 2
+    max_memory = 6.GB
+    max_time   = 3.h
+    reads      = 1e5
+    seed       = 123
+}
+
+process {
+    cpus = 2
+    memory = 3.GB
+    time = 2.h
+
+    publishDir = [
+        path: { "${params.outdir}/${task.process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()}" },
+        mode: 'copy',
+        saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+    ]
+
+    withName: GUNZIP {
+        publishDir  = [
+            enabled: false
+        ]
+    }
+    withName: GUNZIP2 {
+        publishDir  = [
+            enabled: false
+        ]
+    }
+    withName: CHROMSIZES {
+        publishDir  = [
+            enabled: false
+        ]
+    }
+    withName: COOLER_DIGEST {
+        publishDir  = [
+            enabled: false
+        ]
+    }
+    withName: ASSIGN_TYPE {
+        ext.args    = "--count_cutoff 4 --ratio_cutoff 2 --fdr 1"
+        publishDir  = [
+            path: { "${params.outdir}/HiPeak" }
```

```suggestion
            path: "${params.outdir}/HiPeak"
```
The closure is only needed when you want to access something in task context

---

After some testing, I decide to keep the closure. It will work for change the setting from command line but will not work for changing the setting by configuration file. 

For example, 

`nextflow run nf-core/hicar -profile test_full --outdir output // this one work`

`nextflow run nf-core/hicar -profile test_full -c addition.conf //this one will not work`
whith addition.conf:
params {
   outdir     = "output"
}

---

Hmm. I forget people keep wanting to supply parameters this way. I think I should submit a change request so it's more inline with what people think should happen.